### PR TITLE
Capture core dump for z/os in test compilation

### PR DESCRIPTION
**Pull request by JennyChenyj**
- Added moveTDUMP to move core dump to
REPORTDIR/compile

Signed-off-by: Jenny Chen <Jenny.Chen@ibm.com>

### CODE REVIEW

**Comment by JennyChenyj** [comment]
Hey @llxia and @renfeiw! Thanks for reviewing it!

File: makefile [hunk 1]
```diff
@@ -70,5 +70,15 @@ getdependency:
 #######################################
 # compile all tests under $(TEST_ROOT)
 #######################################
+log=$(TESTOUTPUT)$(D)compile$(D)compilation.log
```
Comment (renfeiw) [inline makefile]: please rename `log` to be more specific, something like `COMPILATION_LOG` will be better

File: makefile [hunk 2]
```diff
@@ -70,5 +70,15 @@ getdependency:
...
+ifneq (,$(findstring zos,$(SPEC)))
+    tdumpScript=perl scripts$(D)moveDmp.pl --compileLogPath=$(Q)$(log)$(Q)
+    moveTDump=if [ -z $$(tail -n 1 $(log) | grep 0) ]; then $(tdumpScript); fi
+endif
+
 compile: getdependency
-	ant -f scripts$(D)build_test.xml -DTEST_ROOT=$(TEST_ROOT) -DBUILD_ROOT=$(BUILD_ROOT) -DJDK_VERSION=$(JDK_VERSION) -DJDK_IMPL=$(JDK_IMPL) -DJCL_VERSION=$(JCL_VERSION) -DBUILD_LIST=${BUILD_LIST} -DRESOURCES_DIR=${RESOURCES_DIR} -DPLATFORM=${PLATFORM} -DTEST_JDK_HOME=${TEST_JDK_HOME} -DJVM_VERSION=$(JVM_VERSION)
+	$(MKTREE) $(REPORTDIR)
+	( ant -f scripts$(D)build_test.xml -DTEST_ROOT=$(TEST_ROOT) -DBUILD_ROOT=$(BUILD_ROOT) -DJDK_VERSION=$(JDK_VERSION) -DJDK_IMPL=$(JDK_IMPL) -DJCL_VERSION=$(JCL_VERSION) -DBUILD_LIST=${BUILD_LIST} -DRESOURCES_DIR=${RESOURCES_DIR} -DPLATFORM=${PLATFORM} -DTEST_JDK_HOME=${TEST_JDK_HOME} -DJVM_VERSION=$(JVM_VERSION) 2>&1; echo $$? ) | tee -a $(Q)$(log)$(Q)
+	$(moveTDump)
```
Comment (renfeiw) [inline makefile]: please add a line to remove compilation log

**Comment by JennyChenyj** [comment]
The change was tested:
https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/Grinder/3647/console
Thanks for checking! @renfeiw

File: makefile [hunk 3]
```diff
@@ -70,5 +70,16 @@ getdependency:
...
+COMPILE_CMD=ant -f scripts$(D)build_test.xml -DTEST_ROOT=$(TEST_ROOT) -DBUILD_ROOT=$(BUILD_ROOT) -DJDK_VERSION=$(JDK_VERSION) -DJDK_IMPL=$(JDK_IMPL) -DJCL_VERSION=$(JCL_VERSION) -DBUILD_LIST=${BUILD_LIST} -DRESOURCES_DIR=${RESOURCES_DIR} -DPLATFORM=${PLATFORM} -DTEST_JDK_HOME=${TEST_JDK_HOME} -DJVM_VERSION=$(JVM_VERSION)
+
 compile: getdependency
-	ant -f scripts$(D)build_test.xml -DTEST_ROOT=$(TEST_ROOT) -DBUILD_ROOT=$(BUILD_ROOT) -DJDK_VERSION=$(JDK_VERSION) -DJDK_IMPL=$(JDK_IMPL) -DJCL_VERSION=$(JCL_VERSION) -DBUILD_LIST=${BUILD_LIST} -DRESOURCES_DIR=${RESOURCES_DIR} -DPLATFORM=${PLATFORM} -DTEST_JDK_HOME=${TEST_JDK_HOME} -DJVM_VERSION=$(JVM_VERSION)
+ifneq (,$(findstring zos,$(SPEC)))
+	$(MKTREE) $(TESTOUTPUT)$(D)compile;
+	($(COMPILE_CMD) 2>&1; echo $$? ) | tee $(Q)$(COMPILATION_LOG)$(Q); $(MOVE_TDUMP)
+else
+	$(COMPILE_CMD)
+endif
```
Comment (llxia) [inline makefile]: Talked with @JennyChenyj and @renfeiw , we also need to capture cores for other platforms. Please remove the if in this make file. For simplicity, the logic should be in `moveDmp.pl`.

Comment (llxia) [reply makefile]: Due to the time constraint, we will deliver the zos support first.

**Comment by llxia** [comment]
Due to the time constraint, we will deliver the zos support first.

Tested using PR builds in https://github.com/eclipse/openj9/pull/7707

**Pull request merged**